### PR TITLE
DAOS-14649 test: Skip stages with no tests

### PR DESCRIPTION
**Pull request by brianjmurrell**
### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Comment by brianjmurrell** [comment]
Hopefully this won't need re-reviewing once it's rebased on master after #11702 lands.

**Review by JohnMalmberg — DISMISSED** [review]

**Comment by brianjmurrell** [comment]
Could I please get some reviews on this one?  We have a PR waiting to run testing in GHA and getting these fixes landed before we run it would be good so that it's running accurate testing.

File: .github/workflows/rpm-build-and-test.yml [hunk 1]
```diff
@@ -231,8 +231,8 @@ jobs:
       - name: Checkout code
         uses: actions/checkout@v3
         with:
-          submodules: true
-          fetch-depth: 500
+          submodules: false
+          fetch-depth: 1
```
Comment (ashleypittman) [inline .github/workflows/rpm-build-and-test.yml:235]: This is the default isn't it?

Comment (ashleypittman) [reply .github/workflows/rpm-build-and-test.yml:235]: If the defaults change then the version number changes, it was changing from 2 to 3 that made the default fetch-depth 1.

Comment (brianjmurrell) [reply .github/workflows/rpm-build-and-test.yml:235]: Updated.

**Review by ashleypittman — APPROVED** [review]

**Review by ryon-jensen — APPROVED** [review]

**Pull request merged**